spack test run superlu fails (possibly missing library slu_ddefs.h)

```bash
$ spack test run superlu
==> Spack test a3ovgo5kf4aqeurc3gvruduoiwot63qf
==> Testing package superlu-5.3.0-glyckr7
==> Error: ProcessError: Command exited with status 2:
    'make' 'superlu'

2 errors found in test log:
  >> 5    superlu.c:21:10: fatal error: slu_ddefs.h: No such file or directory
     6       21 | #include "slu_ddefs.h"
     7          |          ^~~~~~~~~~~~~
     8    compilation terminated.
  >> 9    make: *** [Makefile:185: superlu.o] Error 1

See test log for details:
  /home/vanessa/.spack/test/a3ovgo5kf4aqeurc3gvruduoiwot63qf/superlu-5.3.0-glyckr7-test-out.txt

==> Error: 1 test(s) in the suite failed.
```

I'm not sure if this is related but the name is the same so I'll put it here - another issue for superlu-dist
```bash
$ spack test run superlu-dist
==> Spack test 2tcidwyabkv67sbuuvpxe6kcm5pqtelo
==> Testing package superlu-dist-7.1.1-kh6czr3
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8' 'pddrive'

1 error found in test log:
     2    ==> [2021-12-13-17:28:20.864569] Installing /home/vanessa/Desktop/Code/spack-vsoch/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/superlu-dist-7.
          1.1-kh6czr3sbrip2u2kjg32k5cu3mevo7bs/.spack/test to /home/vanessa/.spack/test/2tcidwyabkv67sbuuvpxe6kcm5pqtelo/superlu-dist-7.1.1-kh6czr3/cache/
          superlu-dist
     3    ==> [2021-12-13-17:28:20.887163] FILTER FILE: /home/vanessa/Desktop/Code/spack-vsoch/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/superlu-dist-
          7.1.1-kh6czr3sbrip2u2kjg32k5cu3mevo7bs/.spack/test/make.inc [replacing "^(DSUPERLULIB.+)SRC(.+)"]
     4    ==> [2021-12-13-17:28:20.887723] FILTER FILE: /home/vanessa/Desktop/Code/spack-vsoch/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/superlu-dist-
          7.1.1-kh6czr3sbrip2u2kjg32k5cu3mevo7bs/.spack/test/make.inc [replacing "^LIBS.+\+=.+"]
     5    ==> [2021-12-13-17:28:20.948754] FILTER FILE: /home/vanessa/Desktop/Code/spack-vsoch/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/superlu-dist-
          7.1.1-kh6czr3sbrip2u2kjg32k5cu3mevo7bs/.spack/test/make.inc [replacing "^LIBS[^\+]+=.+"]
     6    ==> [2021-12-13-17:28:20.949972] FILTER FILE: /home/vanessa/Desktop/Code/spack-vsoch/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/superlu-dist-
          7.1.1-kh6czr3sbrip2u2kjg32k5cu3mevo7bs/.spack/test/make.inc [replacing "^LOADOPTS.+"]
     7    ==> [2021-12-13-17:28:20.950512] 'make' '-j8' 'pddrive'
  >> 8    make: *** No rule to make target '/home/vanessa/Desktop/Code/spack-vsoch/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/superlu-dist-7.1.1-kh6czr
          3sbrip2u2kjg32k5cu3mevo7bs//home/vanessa/Desktop/Code/spack-vsoch/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/superlu-dist-7.1.1-kh6czr3sbrip2
          u2kjg32k5cu3mevo7bs/lib/libsuperlu_dist.so', needed by 'pddrive'.  Stop.

See test log for details:
  /home/vanessa/.spack/test/2tcidwyabkv67sbuuvpxe6kcm5pqtelo/superlu-dist-7.1.1-kh6czr3-test-out.txt

==> Error: 1 test(s) in the suite failed.
```

--- a/var/spack/repos/builtin/packages/superlu/package.py
+++ b/var/spack/repos/builtin/packages/superlu/package.py
@@ -115,13 +115,9 @@ class Superlu(CMakePackage):
     def cache_test_sources(self):
         """Copy the example source files after the package is installed to an
         install test subdirectory for use during `spack test run`."""
-        if self.version == Version('5.2.2'):
+        if self.spec.satisfies('@5.2.2:'):
             # Include dir was hardcoded in 5.2.2
             filter_file(r'INCLUDEDIR  = -I\.\./SRC',
-                        'INCLUDEDIR = -I' + self.prefix.include,
-                        join_path(self.examples_src_dir, 'Makefile'))
-
-        filter_file(r'INCLUDEDIR  = -I\.\./SRC',
                         'INCLUDEDIR = -I' + self.prefix.include,
                         join_path(self.examples_src_dir, 'Makefile'))
 

--- a/var/spack/repos/builtin/packages/superlu/package.py
+++ b/var/spack/repos/builtin/packages/superlu/package.py
@@ -172,14 +172,17 @@ class Superlu(CMakePackage):
         return config_args
 
     def run_superlu_test(self, test_dir, exe, args):
-        self.run_test('make',
-                      options=args,
-                      purpose='test: compile {0} example'.format(exe),
-                      work_dir=test_dir)
+        if not self.run_test('make',
+                             options=args,
+                             purpose='test: compile {0} example'.format(exe),
+                             work_dir=test_dir):
+            tty.warn('Skipping test: failed to compile example')
+            return
 
-        self.run_test(exe,
-                      purpose='test: run {0} example'.format(exe),
-                      work_dir=test_dir)
+        if not self.run_test(exe,
+                             purpose='test: run {0} example'.format(exe),
+                             work_dir=test_dir):
+            tty.warn('Skipping test: failed to run example')
 
     def test(self):
         config_args = self._generate_make_hdr_for_test()

--- a/var/spack/repos/builtin/packages/superlu/package.py
+++ b/var/spack/repos/builtin/packages/superlu/package.py
@@ -3,6 +3,8 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import os
+
 
 class Superlu(CMakePackage):
     """SuperLU is a general purpose library for the direct solution of large,
@@ -171,7 +173,8 @@ class Superlu(CMakePackage):
         config_args = self._generate_make_hdr_for_test()
 
         # Write configuration options to make.inc file
-        make_file_inc = join_path(self.install_test_root, self.make_hdr_file)
+        make_file_inc = join_path(self.test_suite.current_test_cache_dir,
+                                  self.make_hdr_file)
         with open(make_file_inc, 'w') as inc:
             for option in config_args:
                 inc.write('{0}\n'.format(option))
@@ -181,9 +184,17 @@ class Superlu(CMakePackage):
             args.append('HEADER=' + self.prefix.include)
         args.append('superlu')
 
-        test_dir = join_path(self.install_test_root, self.examples_src_dir)
+        test_dir = join_path(self.test_suite.current_test_cache_dir,
+                             self.examples_src_dir)
+
+        if not os.path.exists(test_dir):
+            print('Skipping superlu test')
+            return
+
         with working_dir(test_dir, create=False):
             make(*args, parallel=False)
-            self.run_test('./superlu', purpose='Smoke test for superlu',
+            self.run_test('./superlu',
+                          options=['-I{0}'.format(join_path(self.prefix, 'include'))],
+                          purpose='Smoke test for superlu',
                           work_dir='.')
             make('clean')

--- a/var/spack/repos/builtin/packages/superlu/package.py
+++ b/var/spack/repos/builtin/packages/superlu/package.py
@@ -171,13 +171,9 @@ class Superlu(CMakePackage):
 
         return config_args
 
-    def run_superlu_test(self, test_dir, exe):
-        self.run_test(os.environ['CC'],
-                      options=['-I{0}'.format(self.prefix.include),
-                               '-L{0}'.format(self.prefix.lib),
-                               '-L{0}'.format(self.spec['blas'].prefix.lib),
-                               join_path(test_dir, 'superlu.c'), '-o', exe,
-                               '-lsuperlu', '-lopenblas'],
+    def run_superlu_test(self, test_dir, exe, args):
+        self.run_test('make',
+                      options=args,
                       purpose='test: compile {0} example'.format(exe),
                       work_dir=test_dir)
 
@@ -202,17 +198,11 @@ class Superlu(CMakePackage):
 
         test_dir = join_path(self.test_suite.current_test_cache_dir,
                              self.examples_src_dir)
+        exe = 'superlu'
 
-        if not os.path.exists(test_dir):
-            tty.warn('Skipping superlu test: missing required {0}'.format(test_dir))
+        if not os.path.isfile(join_path(test_dir, '{0}.c'.format(exe))):
+            tty.warn('Skipping superlu test:'
+                     'missing file {0}.c'.format(exe))
             return
 
-        #self.run_superlu_test(test_dir, 'superlu')
-        #return
-
-        with working_dir(test_dir, create=False):
-            make(*args, parallel=False)
-            self.run_test('./superlu',
-                          purpose='Smoke test for superlu',
-                          work_dir='.')
-            make('clean')
+        self.run_superlu_test(test_dir, exe, args)

--- a/var/spack/repos/builtin/packages/superlu/package.py
+++ b/var/spack/repos/builtin/packages/superlu/package.py
@@ -5,6 +5,8 @@
 
 import os
 
+from llnl.util import tty
+
 
 class Superlu(CMakePackage):
     """SuperLU is a general purpose library for the direct solution of large,
@@ -177,7 +179,7 @@ class Superlu(CMakePackage):
         self.run_test(os.environ['CC'],
                       options=['-I{0}'.format(self.prefix.include),
                                '-L{0}'.format(self.prefix.lib),
-                               '-L{0}'.format(self.spec['openblas'].prefix.lib),
+                               '-L{0}'.format(self.spec['blas'].prefix.lib),
                                join_path(test_dir, 'superlu.c'), '-o', exe,
                                '-lsuperlu', '-lopenblas'],
                       purpose='test: compile {0} example'.format(exe),
@@ -201,13 +203,12 @@ class Superlu(CMakePackage):
         if self.version < Version('5.2.2'):
             args.append('HEADER=' + self.prefix.include)
         args.append('superlu')
-        args.append('-L{0}'.format(self.spec['openblas'].prefix.lib))
 
         test_dir = join_path(self.test_suite.current_test_cache_dir,
                              self.examples_src_dir)
 
         if not os.path.exists(test_dir):
-            print('Skipping superlu test: missing required {0}'.format(test_dir))
+            tty.warn('Skipping superlu test: missing required {0}'.format(test_dir))
             return
 
         self.run_superlu_test(test_dir, 'superlu')
@@ -216,7 +217,6 @@ class Superlu(CMakePackage):
         with working_dir(test_dir, create=False):
             make(*args, parallel=False)
             self.run_test('./superlu',
-                          options=[],
                           purpose='Smoke test for superlu',
                           work_dir='.')
             make('clean')

--- a/var/spack/repos/builtin/packages/superlu/package.py
+++ b/var/spack/repos/builtin/packages/superlu/package.py
@@ -136,9 +136,9 @@ class Superlu(CMakePackage):
             'ARCH       = ar',
             'ARCHFLAGS  = cr',
             'RANLIB     = {0}'.format('ranlib' if which('ranlib') else 'echo'),
-            'CC         = {0}'.format('cc'),
-            'FORTRAN    = {0}'.format('fc'),
-            'LOADER     = {0}'.format('cc'),
+            'CC         = {0}'.format(env['CC']),
+            'FORTRAN    = {0}'.format(env['FC']),
+            'LOADER     = {0}'.format(env['CC']),
             'CFLAGS     = -O3 -DNDEBUG -DUSE_VENDOR_BLAS -DPRNTlevel=0 -DAdd_',
             'NOOPTS     = -O0'
         ])
@@ -162,9 +162,9 @@ class Superlu(CMakePackage):
             'ARCH       = ar',
             'ARCHFLAGS  = cr',
             'RANLIB     = {0}'.format('ranlib' if which('ranlib') else 'echo'),
-            'CC         = {0}'.format(self.compiler.cc),
-            'FORTRAN    = {0}'.format(self.compiler.fc),
-            'LOADER     = {0}'.format(self.compiler.cc),
+            'CC         = {0}'.format(env['CC']),
+            'FORTRAN    = {0}'.format(env['FC']),
+            'LOADER     = {0}'.format(env['CC']),
             'CFLAGS     = -O3 -DNDEBUG -DUSE_VENDOR_BLAS -DPRNTlevel=0 -DAdd_',
             'NOOPTS     = -O0'
         ])

--- a/var/spack/repos/builtin/packages/superlu/package.py
+++ b/var/spack/repos/builtin/packages/superlu/package.py
@@ -119,6 +119,10 @@ class Superlu(CMakePackage):
                         'INCLUDEDIR = -I' + self.prefix.include,
                         join_path(self.examples_src_dir, 'Makefile'))
 
+        filter_file(r'INCLUDEDIR  = -I\.\./SRC',
+                        'INCLUDEDIR = -I' + self.prefix.include,
+                        join_path(self.examples_src_dir, 'Makefile'))
+
         self.cache_extra_test_sources(self.examples_src_dir)
 
     def _generate_make_hdr_for_test(self):
@@ -169,6 +173,20 @@ class Superlu(CMakePackage):
 
         return config_args
 
+    def run_superlu_test(self, test_dir, exe):
+        self.run_test(os.environ['CC'],
+                      options=['-I{0}'.format(self.prefix.include),
+                               '-L{0}'.format(self.prefix.lib),
+                               '-L{0}'.format(self.spec['openblas'].prefix.lib),
+                               join_path(test_dir, 'superlu.c'), '-o', exe,
+                               '-lsuperlu', '-lopenblas'],
+                      purpose='test: compile {0} example'.format(exe),
+                      work_dir=test_dir)
+
+        self.run_test(exe,
+                      purpose='test: run {0} example'.format(exe),
+                      work_dir=test_dir)
+
     def test(self):
         config_args = self._generate_make_hdr_for_test()
 
@@ -183,6 +201,7 @@ class Superlu(CMakePackage):
         if self.version < Version('5.2.2'):
             args.append('HEADER=' + self.prefix.include)
         args.append('superlu')
+        args.append('-L{0}'.format(self.spec['openblas'].prefix.lib))
 
         test_dir = join_path(self.test_suite.current_test_cache_dir,
                              self.examples_src_dir)
@@ -191,10 +210,13 @@ class Superlu(CMakePackage):
             print('Skipping superlu test: missing required {0}'.format(test_dir))
             return
 
+        self.run_superlu_test(test_dir, 'superlu')
+        return
+
         with working_dir(test_dir, create=False):
             make(*args, parallel=False)
             self.run_test('./superlu',
-                          options=['-I{0}'.format(self.prefix.include)],
+                          options=[],
                           purpose='Smoke test for superlu',
                           work_dir='.')
             make('clean')

--- a/var/spack/repos/builtin/packages/superlu/package.py
+++ b/var/spack/repos/builtin/packages/superlu/package.py
@@ -194,7 +194,7 @@ class Superlu(CMakePackage):
         with working_dir(test_dir, create=False):
             make(*args, parallel=False)
             self.run_test('./superlu',
-                          options=['-I{0}'.format(join_path(self.prefix, 'include'))],
+                          options=['-I{0}'.format(self.prefix.include)],
                           purpose='Smoke test for superlu',
                           work_dir='.')
             make('clean')

--- a/var/spack/repos/builtin/packages/superlu/package.py
+++ b/var/spack/repos/builtin/packages/superlu/package.py
@@ -136,9 +136,9 @@ class Superlu(CMakePackage):
             'ARCH       = ar',
             'ARCHFLAGS  = cr',
             'RANLIB     = {0}'.format('ranlib' if which('ranlib') else 'echo'),
-            'CC         = {0}'.format(self.compiler.cc),
-            'FORTRAN    = {0}'.format(self.compiler.fc),
-            'LOADER     = {0}'.format(self.compiler.cc),
+            'CC         = {0}'.format('cc'),
+            'FORTRAN    = {0}'.format('fc'),
+            'LOADER     = {0}'.format('cc'),
             'CFLAGS     = -O3 -DNDEBUG -DUSE_VENDOR_BLAS -DPRNTlevel=0 -DAdd_',
             'NOOPTS     = -O0'
         ])
@@ -207,8 +207,8 @@ class Superlu(CMakePackage):
             tty.warn('Skipping superlu test: missing required {0}'.format(test_dir))
             return
 
-        self.run_superlu_test(test_dir, 'superlu')
-        return
+        #self.run_superlu_test(test_dir, 'superlu')
+        #return
 
         with working_dir(test_dir, create=False):
             make(*args, parallel=False)

--- a/var/spack/repos/builtin/packages/superlu/package.py
+++ b/var/spack/repos/builtin/packages/superlu/package.py
@@ -188,7 +188,7 @@ class Superlu(CMakePackage):
                              self.examples_src_dir)
 
         if not os.path.exists(test_dir):
-            print('Skipping superlu test')
+            print('Skipping superlu test: missing required {0}'.format(test_dir))
             return
 
         with working_dir(test_dir, create=False):

